### PR TITLE
all issues solved

### PR DESCRIPTION
**Pull request by mayank33**

### CODE REVIEW

**Comment by epifanio** [comment]
@mayank33 @rashadkm can we set  [those envs](https://github.com/rashadkm/webgrass/pull/14/files#diff-85af83c30b8d57c5c46e89012857294fR181) in the cmake settings?

**Comment by rashadkm** [comment]
not in the cmake settings. that will say that once compiled and run there is no change.

like we discussed, we put them in a shell script inside webgrass/scripts and call it init_grass.sh.

Then later we can have an option in gui to change via browser. That will not be exactly updating the script but rather using wt and grass call to change the mapset, location, output image format etc..

**Comment by rashadkm** [comment]
i will test this on epinux tomorrow and merge it

**Comment by rashadkm** [comment]
Add the env vars in the script and put the script in a "scripts" dir. And remove the export calls in the pstream call.

**Comment by mayank33** [comment]
done

File: src/MainUI.cpp [hunk 1]
```diff
@@ -157,57 +143,59 @@ void MainUI::openModuleUI(Wt::WMenuItem* gitem) {
...
+
+                 }
+        std::cout<<command<<std::endl;
+            // redi::ipstream proc("grass70 $HOME/grassdata/newLocation/PERMANENT", redi::pstreams::pstdout | redi::pstreams::pstdin );
+            //   std::string line;
+            //   std::cout.flush();
+            //   while (std::getline(proc.out(), line)){
+            //     std::cout << "stdout: " << line << '\n';
+            // }
+            redi::ipstream proc("export GRASS_PNG_AUTO_WRITE=TRUE; export GRASS_PNG_COMPRESSION=9; export GRASS_TRANSPARENT=TRUE; export GRASS_TRUECOLOR=TRUE; export LD_LIBRARY_PATH=/usr/lib/grass70/lib; export GISBASE=/usr/lib/grass70/; export GISDBASE=/home/mayank/grassdata1; export GISRC=/home/mayank/.grass7/rc; export PATH=/usr/lib/grass70/bin:/usr/lib/grass70/scripts:$PATH; g.gisenv");
```
Comment (rashadkm) [inline src/MainUI.cpp]: what if you do those export in shell script and then running grass commands such as g.list rast ?

Comment (rashadkm) [inline src/MainUI.cpp]: doing export there and then do g.list as a list in ipstream might not work. can you test it please?

Comment (rashadkm) [inline src/MainUI.cpp]: you still could get mapset permission denied. but i want clean this up before merge. still getting mapset permission denied. i would recommend to build grass from source. the worst case, is we patch the source for webgrass!. but before that i want the script to run first and then run this command.

remember to run script as . ./scripts/grass_init.sh. (note the first dot )

**Pull request merged**